Rename menu ID to use hyphens

andybroomfield opened the pull request:
<!-- See https://docs.localgovdrupal.org/contributing/ for guidelines on contributing. -->

Fix #307
Fix #140

## What does this change?

Rename the menu `localgov_services_menu` to use hypens as in `localgov-services-menu`
Need to check if this is actually used in places apart from config and if other modules (localgov_demo) or themes (localgov_base) need to be updated to account for the change.

<!-- A pull request should have enough detail to be understandable far in the
future. e.g what is the problem/why is the change needed, how does it solve it
and any questions or points of discussion. Prefer copying information from a
Trello card (for example) over linking to it; the card may not always exist and
reviewers may not have access to the board. -->

## How to test

Go to /admin/structure/menu/manage/localgov_services_menu and save the menu.
(Requires fresh install).

<!-- Provide instructions to help others verify the change. This could take the
form of "On main, do X and witness Y. On this branch, do X and witness Z. " -->

## How can we measure success?

No more errors when trying to edit the services menu with 

> "The machine-readable name must contain only lowercase letters, numbers, and hyphens."

<!-- Do you expect errors to decrease? Do you expect user journeys to be
simplified? What can be used to prove this? A filtered view of logs or
analytics, etc? -->

## Have we considered potential risks?

Menu name is used in places independently (custom themes).

<!-- What are the potential risks and how can they be mitigated? Does an error
require an alarm? Should user help, infosec, or legal be informed of this
change? Is private information guarded? Do we need to add anything in the
backlog? -->

## Images
n/a

<!-- Usually only applicable to UI changes, what did it look like before and
what will it look like after? -->

## Accessibility
n/a

andybroomfield committed:
Rename menu ID to use hyphens

andybroomfield committed:
Rename system.menu.localgov_services_menu.yml to system.menu.localgov-services-menu.yml

andybroomfield commented:
Here's what we need to update
https://github.com/search?q=org%3Alocalgovdrupal%20localgov_services_menu&type=code

andybroomfield committed:
Fix menu block configuration for localgov services

andybroomfield committed:
Fix menu block configuration for localgov services

andybroomfield commented:
I've updated block configuration here, there are some code changes to Localgov_demo that will need to be carried out and merged at the same time if we decide to go with this.

I couldn't find any references in localgov_base. If I've missed something @markconroy it would be good to check the theme can still work with this change.

I haven't made an update hook. Is that something we require for this change or are we happy it's new sites only. Presumably most who have run into the issue have done what BHCC did and create our own menu a long time ago so it's not relevant.

andybroomfield commented:
It appears as long as the menu name is `localgov-services-menu` then the magic services menu in localgov_base continues to work.
<img width="1233" height="272" alt="Screenshot 2025-11-24 at 3 26 41 pm" src="https://github.com/user-attachments/assets/1a1b591b-43c2-4f74-8cb5-9b358f5c83e7" />

markconroy commented:
@andybroomfield I think I'd like an update hook for this (maybe as a follow up if we wish) but it definitely seems to be a bug that we have that we should fix across the board.

tonypaulbarker commented:
Some thoughts on supporting existing sites.

I’m not sure we can do update hooks because we won’t know the naming for the active theme block config? Is it possible that theming overrides can depend on the underscores too? 

One idea is a snippet to bypass Drupal core’s validation for that specific machine name?

andybroomfield committed:
Add update hook to create hypen seperated services menu and move menu link content

This will
- Create the new menu, `localgov-services-menu`
- Update menu_link_content entities (user created menu items) to set the menu name
  to `localgov-services-menu` from `localgov_services_menu`
- Delete the old menu `localgov_services_menu`

andybroomfield commented:
@markconroy @tonypaulbarker I've pushed an update hook for review. I only think I can update the machine name of the menu by creating a new one and moving the items over. But any other configuration I'm not so sure about in terms of how we go about it. Would we have to discover each block and then try and update them, and would it effect a theme.

finnlewis commented:
I'm also testing.

finnlewis committed:
Fix coding standards.

finnlewis commented:
Just fixed some coding standards so the rest of the tests can run.

finnlewis commented:
Update hook works nicely for me @andybroomfield 

Before: 
<img width="1826" height="1432" alt="image" src="https://github.com/user-attachments/assets/69a0dffd-27fa-433a-ad70-1d843a7e5fbc" />

after: 
<img width="2022" height="1300" alt="image" src="https://github.com/user-attachments/assets/eedeb6f6-785d-4823-b1e7-f0d954cf09eb" />

finnlewis reviewed (approved):
Works for me! :)

finnlewis commented:
Discussing in Merge Tuesday: 

For existing sites block placements is an issue, attempting to delete the old menu will remove the block placement.

Andy suggests we create the menu, copy links to the new menu but don's delete the old menu.
 
Add rerelease notes to be clear that developers need to:

 1. remove the old menu block, place the new menu block. 
 2. update any other code that references the menu block name.
 3. delete the old menu

andybroomfield committed:
Relabel the old (broken) localgov_services_menu instead of deleting it

In order to prevent existing sites breaking
- Check if an existing `localgov-services-menu` exists, and abort if so.
- Duplicate menu content items instead of moving them.
- Relabel the existing `localgov_services_menu` indicating it should be
  deleted once the blocks have been replaced.

AWearring reviewed (commented):
Tested on the West Lindsey build where we had already "fixed" the issue by renaming the menu via a patch. `localgov_services` happily updated and the update hook gets skipped because of `if ($existing)`.

Will try testing on a new LGD vanilla build later

AWearring reviewed (approved):
Have no tested on an old LGD vanilla build I had locally and the update hook correctly copies the Services Menu and updates the existing menu to make it obvious that it should be removed
<img width="1122" height="114" alt="image" src="https://github.com/user-attachments/assets/6914e2f9-f974-4286-aa82-2b04004fa812" />

andybroomfield commented:
This PR has been recreated on Drupal.org
https://git.drupalcode.org/project/localgov_services/-/merge_requests/3
Please merge there and close this PR.

andybroomfield commented:
Should we close the PR here as merged on Drupal.org?

finnlewis commented:
This is merged on drupal.org https://git.drupalcode.org/project/localgov_services/-/merge_requests/3

Cosing here
